app/src/main/java/com/github/jorgecastillo/kotlinandroid/data/HeroesRepository.kt

```diff
@@ -2,7 +2,11 @@ package com.github.jorgecastillo.kotlinandroid.data
 
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchAllHeroes
 import com.github.jorgecastillo.kotlinandroid.data.datasource.remote.fetchHeroesFromAvengerComics
+import com.karumi.marvelapiclient.model.CharacterDto
+import kategory.HK
 
-fun getHeroesWithCachePolicy() = fetchAllHeroes()
+inline fun <reified F> getHeroesWithCachePolicy(): HK<F, List<CharacterDto>> =
```

Comment:
why do you need to make those functions inline and reified exactly? that might be a problem in terms of `kategory` usage. The user should never need to know how our implicit implementation is done behind the scenes (if possible), and seems like the reified here is just for that. Am I wrong?

If you look at the function implementation, there is nothing telling you that the `F` type needs to be reified, which is gonna be clearly confusing for the users of the lib / readers of this code. I feel like the usage of implicits was much more transparent to the user before the later changes on the global instances system.